src/types/shift.cr

```diff
@@ -69,5 +72,17 @@ module Tanda::CLI
     def total_break_minutes : Time::Span
       breaks.sum(&.length).minutes
     end
+
+    def leave? : Bool
+      !leave_request_id.nil?
+    end
+
+    def leave_request
+      leave_id = leave_request_id
+      return if leave_id.nil?
+
+      client = Current.client!
```

Comment:
Worth thinking of a way to not have the client in a global singleton for this use-case